### DISCUSSION TITLE
Combined Joint Range of Series Connected Joints

### DISCUSSION THREAD

**Discussion by juhorsch** [google-deepmind/mujoco#2235] [Asking for Help] [answered]
### Intro

Hi!

I am a graduate student at DLR, I use MuJoCo for my research on series elastic actuators.

### My setup

Mujoco MJX used in a brax env, Python

### My question

I am trying to implement a series elastic actuator setup, by attaching two joints (motor and spring) in series:

```<body name="left_upper_shank" pos="0.059999999999999998 -0.175 -0.275" euler="0 0 0">
          <!-- <body name="left_upper_shank" pos="0.059999999999999998 -0.125 -0.315" euler="0 0 0"> -->
            <inertial pos="0.055127279000000001 -0.02463694 -0.13230046000000001" mass="5.8600000000000003" fullinertia="0.092326000000000005 0.10798000000000001 0.038918000000000001 -0.0032339999999999999 -0.013306999999999999 0.00029631999999999998"/>
            <geom name="left_upper_shank_visual" type="mesh" group="0" pos="0 0 0" euler="0 0 0" mesh="cl_upper_leg_scaled_reduced"/>
            <joint name="left_hip_motor" pos="0 0 0" axis="0 1 0" class="default_joint"/>
            <joint name="left_hip_spring" stiffness="429" damping="6.4" class="default_spring"/>
```

Is there a clever way of setting a range for the combined angle? Basically the motor link really has no limit, however the leg has an endstop with range="-80.214091318315241 28.647889756541161".

Thanks in advance

### Confirmations

- [X] I searched the [latest documentation](https://mujoco.readthedocs.io/en/latest/overview.html) thoroughly before posting.
- [X] I searched previous [Issues](https://github.com/google-deepmind/mujoco/issues) and [Discussions](https://github.com/google-deepmind/mujoco/discussions), I am certain this has not been raised before.

**Comment by yuvaltassa** [comment, marked as answer]
Yes. You can define a fixed tendon whose length is the sum of the joint angles, and make that tendon have range limits.

**Reply by juhorsch** [reply]
Oh wow yeah - that was easy! Thank you so much!
